go/tasks/plugins/k8s/sagemaker/sagemaker.go

```diff
@@ -282,9 +384,10 @@ func (m awsSagemakerPlugin) BuildResourceForHyperparameterTuningJob(
 	// in the static map and let the one in the map of the tunable hyperparameters take precedence
 	staticHyperparams = deleteConflictingStaticHyperparameters(ctx, staticHyperparams, hpoJobConfig.GetHyperparameterRanges().GetParameterRangeMap())
 
-	taskName := taskCtx.TaskExecutionMetadata().GetTaskExecutionID().GetID().NodeExecutionId.GetExecutionId().GetName()
+	// taskName := taskCtx.TaskExecutionMetadata().GetTaskExecutionID().GetID().NodeExecutionId.GetExecutionId().GetName()
+	jobName := taskCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName()
 
-	trainingImageStr, err := getTrainingImage(ctx, sagemakerHPOJob.GetTrainingJob())
+	trainingImageStr, err := getTrainingJobImage(ctx, taskCtx, sagemakerHPOJob.GetTrainingJob()) // TODO: replace this
```

Comment:
Replace this with what?

Reply:
This comment does not apply to the new version of code